spec/unit/shoulda/matchers/active_model/validate_numericality_of_matcher_spec.rb

```diff
@@ -1113,6 +1113,52 @@ def set_attr!; self.attr = 5 end
     end
   end
 
+  context 'when the writer method for the attribute changes incoming values' do
+    context 'and the matcher knows nothing of this' do
+      it 'raises a CouldNotSetAttributeError' do
+        model = define_model_validating_numericality(attribute_name: :zip_code)
+
+        model.class_eval do
+          def zip_code=(name)
```

Comment:
Unused method argument - `name`. If it's necessary, use `_` or `_name` as an argument name to indicate that it won't be used. You can also write as `zip_code=(*)` if you want the method to accept any arguments but don't care about them.